### PR TITLE
[Aptos StdLib] Fix potential overflow issues with math average

### PR DESCRIPTION
**Pull request by movekevin**
### Description

This was found in an audit. (a + b) can overflow.

### Test Plan
Unit tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4270)

### CODE REVIEW

**Review by vgao1996 — APPROVED** [review]

File: aptos-move/framework/aptos-stdlib/sources/math128.move [hunk 1]
```diff
@@ -13,7 +13,11 @@ module aptos_std::math128 {
 
     /// Return the average of two.
     public fun average(a: u128, b: u128): u128 {
-        (a + b) / 2
+        if (a < b) {
```
Comment (wrwg) [inline aptos-move/framework/aptos-stdlib/sources/math128.move:16]: neat

**Review by wrwg — APPROVED** [review]

File: aptos-move/framework/aptos-stdlib/sources/math128.move (hunk 1 above)
Comment (junkil-park) [reply aptos-move/framework/aptos-stdlib/sources/math128.move:16]: Just a note that `ensures result == (a + b) / 2;` is provable as well :)

Comment (movekevin) [reply aptos-move/framework/aptos-stdlib/sources/math128.move:16]: Added a spec statement for average

Comment (lightmark) [reply aptos-move/framework/aptos-stdlib/sources/math128.move:16]: can we do 
```suggestion
       (a & b) + ((a ^ b) >> 1) 
```

Comment (wrwg) [reply aptos-move/framework/aptos-stdlib/sources/math128.move:16]: > can we do

Please not. It forces the prover to use bitvector repr of ints (once this is implemented) which is way slower for arithmetics than this bit fiddling. The gas cost is equivalent.

Comment (wrwg) [reply aptos-move/framework/aptos-stdlib/sources/math128.move:16]: > > can we do
> 
> Please not. It forces the prover to use bitvector repr of ints (once this is implemented) which is way slower for arithmetics. The gas cost is equivalent.

**Pull request merged**